lib/sycamore/sycamore/docset.py

```diff
@@ -771,6 +772,50 @@ def process_doc(doc: Document) -> Document:
 
         return self.map(process_doc, **resource_args)
 
+    def llm_filter(
+        self,
+        client: OpenAI,
+        new_field: str,
+        messages: List[dict],
```

Comment:
This feels a little strange because you're expecting the caller to know how to format generic messages. What do you think about just accepting a prompt here? And that can be messages or single string

Reply:
Hmm, so list[dict] or str? And based on which one it is do {"messages": messages} or {"prompt": prompt}?

Reply:
can just accept a dict like prompt_kwargs (the way extract_entity does), and that can either have messages or prompt

Reply:
I think extract_entity does it in a different way, uses a prompt template